src/position.cpp

```diff
@@ -839,6 +839,9 @@ void Position::do_move(Move m, StateInfo& newSt, bool givesCheck) {
   // Set capture piece
   st->capturedPiece = captured;
 
+  // Draw at first repetition
+  st->draw = true;
```

Comment:
I don't like to set st->draw = true; for each searched node in do_move(), this seems to me a real waste.

Eventually you can use gamePly in the draw condition like:

       if (stp->key == st->key)
           return stp->gamePly > RootPosition->gamePly || stp->draw;